diesel/src/associations/mod.rs

```diff
@@ -122,6 +122,19 @@ use query_source::Table;
 
 pub use self::belongs_to::{BelongsTo, GroupedBy};
 
+/// Represents a struct which can be identified on a single table in the
+/// database. This must be implemented to use associations, and some features of
+/// updating.
+///
+/// ### Deriving
+///
+/// This trait can be automatically derived using `diesel_codegen` by adding
+/// `#[derive(Identifiable)]` to your struct. The primary key will be assumed to
+/// be a field and column called `id`. By default the table will be
+/// assumed to be the plural form of the struct name (using *very* dumb
+/// pluralization). If your table name differs from that convention, or requires
```

Comment:
Imagine you have a struct called `Mouse` and diesel 0.8 assumes your table name is `Mouses` – which it for some reason it actually is. Now, diesel 0.8.2 comes with an awesome new pluralization and infers your table name is `Mice`.  You are also not using `infer_schema` so your code continues to compile. Sadly, your application doesn't work any longer.

We should say whether we guarantee this to never happen.